GVRf/Framework/framework/src/main/java/org/gearvrf/scene_objects/GVRVideoSceneObject.java

```diff
@@ -359,17 +362,22 @@ public long getTimeStamp() {
          * {@link MediaPlayer}
          */
         public void release() {
-            if (mMediaPlayer != null) {
-                mMediaPlayer.release();
-                mMediaPlayer = null;
+            if (mMediaPlayerRef != null) {
```

Comment:
You need to get() and hold on to the reference via a strong one. Theoretically by the time you make the second get() call it may return null.